src/threading.c

```diff
@@ -641,20 +643,35 @@ void jl_init_threading(void)
         }
     }
 
-    jl_all_tls_states_size = nthreads + nthreadsi;
+    int16_t ngcthreads = jl_options.ngcthreads - 1;
```

Comment:
It would be good to have environment variable control over this also, as with `--threads`. Something like `JULIA_NUM_GC_THREADS`?